src/components/Filters/SchemaSieve.ts

```diff
@@ -1,157 +1,516 @@
-		// and don't filter the item
-		return true;
+	return filter as JSONSchema6;
+};
+
+// Update or insert a full text search filter into an array of filters
+export const upsertFullTextSearch = (
+	schema: JSONSchema6,
+	filters: JSONSchema6[] = [],
+	term: string,
+) => {
+	const searchFilter = createFullTextSearchFilter(schema, term);
+	const existingSearchIndex = findIndex(filters, { title: FULL_TEXT_SLUG });
+
+	if (existingSearchIndex > -1) {
+		filters.splice(existingSearchIndex, 1, searchFilter);
+	} else {
+		filters.push(searchFilter as JSONSchema6);
 	}
 
-	filter<T>(
-		collection: T[],
-		input: FilterRule | FilterRule[],
-	): T[] | Partial<T> {
-		if (isArray(collection)) {
-			return this.filterArray(collection, input);
+	return filters;
+};
+
+// Removes a full text search filter from an array of filters
+export const removeFullTextSearch = (filters: JSONSchema6[]) =>
+	filters.filter(f => f.title !== FULL_TEXT_SLUG);
+
+export const createFilter = (
+	schema: JSONSchema6,
+	signatures: FilterSignature[],
+): JSONSchema6 => {
+	const anyOf = signatures.map(({ field, operator, value }) => {
+		const subSchema = schema.properties![field] as JSONSchema6;
+		const model = getDataModel(subSchema);
+
+		if (!model || !model.operators.hasOwnProperty(operator)) {
+			return {};
 		}
-		if (isPlainObject(collection)) {
-			return this.filterObject(collection, input) as Partial<T>;
+
+		return model.createFilter(field, operator as any, value, subSchema);
+	});
+
+	return {
+		$id: utils.randomString(),
+		anyOf,
+	};
+};
+
+export const decodeFilter = (schema: JSONSchema6, filter: JSONSchema6) => {
+	const signatures = filter.anyOf!.map(f => {
+		const schemaField = Object.keys(f.properties!).shift()!;
+		const subSchema = schema.properties![schemaField] as JSONSchema6;
+		const model = getDataModel(subSchema);
+
+		if (!model) {
+			return {};
 		}
 
-		throw new Error('collection argument must be either object or array.');
+		return model.decodeFilter(f);
+	});
+
+	return signatures.filter(s => s !== null);
+};
+
+export const getOperators = (schema: JSONSchema6, field: string) => {
+	const subSchema = schema.properties![field] as JSONSchema6;
+	const model = getDataModel(subSchema);
+
+	if (!model) {
+		return [];
 	}
 
-	filterArray(collection: any[], input: FilterRule | FilterRule[]) {
-		return collection.filter(item => this.test(item, input));
+	return map(model.operators, (value, key) => ({
+		slug: key,
+		label: value.getLabel(subSchema),
+	}));
+};
+
+const convertToBooleanSignature = (
+	legacyFilter: v3.BaseFilter,
+): FilterSignature => {
+	return {
+		field: legacyFilter.name,
+		value: legacyFilter.operator === 'is true',
+		operator: 'is',
+	};
+};
+
+const convertToStringSignature = (
+	legacyFilter: v3.BaseFilter,
+): FilterSignature => {
+	const signature = {
+		field: legacyFilter.name,
+		value: legacyFilter.value,
+		operator: '',
+	};
+
+	switch (legacyFilter.operator) {
```

Comment:
It seems that this switch could be replaced by a mapping object and have something like
```
signature.operator = legacyFilterMap[legacyFilter.operator] || legacyFilter.operator || '';
```
Probably also applies to other methods.